src/afa/afa.cc

```diff
@@ -34,29 +36,531 @@ const std::string Mata::Afa::TYPE_AFA = "AFA";
 
 std::ostream& std::operator<<(std::ostream& os, const Mata::Afa::Trans& trans)
 { // {{{
-	std::string result = "(" + std::to_string(trans.src) + ", " + trans.formula + ")";
+	std::string result = "(" + std::to_string(trans.src) + ", " 
+                       + std::to_string(trans.symb) + ", " 
+                       + std::to_string(trans.dst) + ")";
 	return os << result;
 } // operator<<(ostream, Trans) }}}
 
-
+/** This function adds a new transition to the automaton. It changes a transition
+* relation. 
+* @brief adds a new transition
+* @param trans a given transition
+*/
 void Afa::add_trans(const Trans& trans)
 { // {{{
-	this->transitions.push_back(trans);
+    assert(!this->transRelation.empty());
+
+    // Performs a transition using given src state and symbol
+    // in context of transitions which have been already added to the automaton
+    // If there is no result, a new transition will be created
+    // If the corresponding transition already exists, given 'dst' will be added to
+    // the transition
+    auto nodesPtr = perform_trans(trans.src, trans.symb);
+    if(!nodesPtr->empty())
+    {
+        // TODO: It would be nice to remove redundant clauses (like (1 || (1 && 2)))
+        // The clause (1 && 2) could be deleted. Solution:
+        // create closed set -> add the new transition -> extract the antichain ->
+        // store it to the transition (???). Working solution:
+        //        
+        // auto cl = ClosedSet<State>(ClosedSet<State>::upward_closed, 0, max_state_used, *nodesPtr);
+        // cl.insert(trans.dst);
+        // *nodesPtr = cl.get_antichain();
+        //        
+        // Is it worth it?         
+
+        *nodesPtr = nodesPtr->Union(trans.dst);
+        return;
+    }
+    transRelation[trans.src].push_back(trans);
+
 } // add_trans }}}
 
+
+/** This function adds a new inverse transition to the automaton
+* It changes an inverse transition relation.
+* @brief adds a new inverse transition
+* @param trans a given TRANSITION (it will be inverted within this function)
+*/
+void Afa::add_inv_trans(const Trans& trans)
+{// {{{
+
+    // Iterating through all the nodes which were given as a destination of the current transition
+    for(auto node : trans.dst)
+    {
+        // TODO: Could this ever happen?
+        if(node.empty())
+        {
+            continue;
+        }
+
+        // If there is already a memory cell which corresponds to the current dst node and a transition symbol,
```

Comment:
Could we try to keep the code contained within the 120 characters boundary? I had to scroll to the right in my IDE to read the whole comment, which is really unpleasant.

Reply:
Thanks for your suggestion, I did not realise. I have fixed it.